### PR TITLE
Upgrade specs to 0.8

### PR DESCRIPTION
**Pull request by torkleyy**

### CODE REVIEW

File: src/engine/app.rs [hunk 1]
```diff
@@ -230,10 +235,12 @@ impl<T> ApplicationBuilder<T>
     /// Creates a new ApplicationBuilder with the given initial game state and
     /// display configuration.
     pub fn new(initial_state: T, cfg: DisplayConfig) -> ApplicationBuilder<T> {
+        let pool = Arc::new(ThreadPool::new(num_cpus::get()));
```
Comment (torkleyy) [inline src/engine/app.rs]: I'm going to use this pool in the next PR, which will allow a shared thread pool.

**Comment by kvark** [comment]
Good stuff!
Usage of `join` for non-tuples is weird, there could be a better way, but it's not significant.

**Comment by torkleyy** [comment]
@kvark Yeah, was thinking the same.

**Comment by kvark** [comment]
Also, a squash would be appropriate.

**Comment by torkleyy** [comment]
@kvark Isn't there a way to squash when merging?

**Comment by kvark** [comment]
@torkleyy yep, I'll do that.
Typically, you'd have some sort of a homu/bors bot that doesn't do squashing, so I'm used to request that from time to time ;)

**Pull request merged**